lib/search/packages.ex

```diff
@@ -0,0 +1,189 @@
+defmodule Search.Packages do
+  @moduledoc """
+  The Packages context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Search.Repo
+
+  alias Search.Packages.{Package, DocItem, DocFragment}
+  alias Search.{HexClient, ExDocParser}
+
+  @doc """
+  If given a package name, adds the latest version of the package to the app. If given a `#{HexClient.Release}` adds
+  the specified release. Does not embed it yet.
+  """
+  def add_package(package_name) when is_binary(package_name) do
+    case HexClient.get_releases(package_name) do
+      {:ok, releases} ->
+        latest = Enum.max_by(releases, & &1.version, Version)
```

Comment:
the "latest version" is a bit more subtle, it is this if the only versions are pre-releases, but otherwise it is the latest non-pre-release version.

We have a [Postgrex v1.0.0-rc.1](https://hex.pm/packages/postgrex) released, checks notes, 6 years ago, and that is not considered "latest version".

See https://github.com/hexpm/hexdocs/blob/20a09a705c312aaa2c22a894eaf018468eca50bb/lib/hexdocs/utils.ex#L33:L54